core/modules/replica/WorkerSqlRequest.cc

```diff
@@ -48,56 +52,61 @@ WorkerSqlRequest::Ptr WorkerSqlRequest::create(ServiceProvider::Ptr const& servi
                                                int priority,
                                                std::string const& query,
                                                std::string const& user,
-                                               std::string const& password) {
+                                               std::string const& password,
+                                               size_t maxRows) {
     return WorkerSqlRequest::Ptr(
         new WorkerSqlRequest(serviceProvider,
                              worker,
                              id,
                              priority,
                              query,
                              user,
-                             password));
+                             password,
+                             maxRows));
 }
 
 
 WorkerSqlRequest::WorkerSqlRequest(ServiceProvider::Ptr const& serviceProvider,
-                                     string const& worker,
-                                     string const& id,
-                                     int priority,
-                                     std::string const& query,
-                                     std::string const& user,
-                                     std::string const& password)
+                                   string const& worker,
+                                   string const& id,
+                                   int priority,
+                                   std::string const& query,
+                                   std::string const& user,
+                                   std::string const& password,
+                                   size_t maxRows)
     :   WorkerRequest(serviceProvider,
                       worker,
                       "SQL",
                       id,
                       priority),
         _query(query),
         _user(user),
-        _password(password) {
+        _password(password),
+        _maxRows(maxRows) {
 }
 
 
 void WorkerSqlRequest::setInfo(ProtocolResponseSql& response) const {
 
-    LOGS(_log, LOG_LVL_DEBUG, context() << __func__);
+    LOGS(_log, LOG_LVL_DEBUG, context(__func__));
 
-    util::Lock lock(_mtx, context() + __func__);
+    util::Lock lock(_mtx, context(__func__));
 
-    // Return the performance of the target request
+    // Update the performance of the target request before returning it
+    _response.set_allocated_target_performance(performance().info());
 
-    response.set_allocated_target_performance(performance().info());
-
-    // TODO: implement extracting and setting a result set
-    //response.set_data(data());
+    // Note, we shall not invalidate the local state of the cached response
+    // object because, by a design of the Replication Framework,  the state
```

Comment:
IMO I'd prefer a single space after the comma than have the end of the word `state` line up with the end of the word `response` 🙃

Reply:
This confusing comment was eliminated in a later commit